### ISSUE TITLE
[general] distribute on alternate stores (IzzyOnDroid, FDroid)

### ISSUE THREAD

**Issue by reusityback** [diegoberaldin/RaccoonForLemmy#470]
Would it be possible to distribute Raccoon For Lemmy also on F-Droid?

**Comment by diegoberaldin** [comment]
I've struggled to have it there last autumn but with no success. I've had the feeling that unless your project has enough "momentum" they don't really consider it worth being on their store, so I decided to pause the submission process and restart when the project was big enough. I'll have to remove Firebase Crashlytics but that won't be a big loss 😉😉😉

**Comment by diegoberaldin** [comment]
Just for information, Firebase Crashlytics has been removed in #522, which was one of the main blockers for F-Droid submission.

Now the problem is... getting attention by F-Droid maintainers and dealing with their reproducible build nightmare system. Next time I'll publish a release (so basically when there will be the first tag of a Crashlytics-free build) I'll consider resubmitting to their metadata repository. 

But they have Biblical times for answering and in case builds/checks fail no one will ever help you not even if you pay them, so don't expect news soon.

**Comment by IzzySoft** [comment]
I just took a look: `AD_ID` doesn't seem to come from Racoon itself, so probably some dependency slipped that in (though I have no idea from which). You could try this approach to get rid of it then, as obviously you don't need it yourself: [Removing Unwanted Manifest Permissions With tools:node](https://developermemos.com/posts/removing-unwanted-manifest-permissions/).

As for APK size: native libs seem just to be "stored", not compressed. So going without the two x86 ABIs (which are quite unusual on Android devices and thus mostly relevant for emulators) would mean reducing APK size by about 10 MB. Going for a single ARM one would save another 5 MB, so I could keep 2 versions of that while staying inside the 30 MB limit.

**Comment by IzzySoft** [comment]
Well, maybe better get rid of Firebase and GMS. What do you need it for (so I maybe can name a few alternatives/replacements)? If it's for Push Messages, I'd recommend taking a look at @UnifiedPush – for all else there're [appwrite](https://github.com/appwrite/appwrite) and [Supabase](https://supabase.com/).

And yes, I was referring to LiftOff as well, which was in my repo until recently.

**Comment by diegoberaldin** [comment]
I got rid of Firebase (Crashlytics was the reason why I added the AD_ID permission in the beginning) but I accidentally forgot the permission which was removed only in the latest tag.

If you need me to create a new stable version other than 1.8.0-20240229-pre I can do it, and I can also release it to the Play store to keep versions on sync.

**Comment by IzzySoft** [comment]
If you'd had such a build, ideally with a per-ABI (I'd suggest the arm64), I can set it up here. Having it in sync with Play is no requirement for me (it paints a smile on my face when my repo has the newer version and Play is behind :speak_no_evil: :dash:). Just let me know when it's there, and I see to get things rolling.

**Comment by diegoberaldin** [comment]
Done in the [latest release](https://github.com/diegoberaldin/RaccoonForLemmy/releases/tag/1.8.0)!

Thank you Izzy 🙏 : this is still a very small project and having it on your store will help it grow in adoption, plus for me personally it is a dream becoming true. 🤩

**Comment by IzzySoft** [comment]
Thanks @diegoberaldin – setting it up now. Please remember to include the lines to get rid of that dependency info blob, which is still there with the new APK.

If you don't mind a few suggestions for the app description: those **using** the app are less interested in the technologies you use (Kotlin, Jetpack Compose) but more in the features it offers. So if I might suggest a replacement for the `en-US` `full_description.txt` (also to be adapted for the `it-IT` one then), as taken from the README:

```html
<p><i>Raccoon for Lemmy</i> is fully open source, and intends to offer an ad-free and as functional as possible experience on Lemmy.</p>
<p>Concerning customization, the ability to change some aspects like font face or size and app colors, vote format, bar transparency and so on was of paramount importance from the very beginning. Similarly, users should be able to use the app in their native language and change the UI language independently from the system language.</p>
<p>This app is also intended for moderators who want to use their mobile device, offering moderation tools (feature post, lock post, distinguish comment, remove post/comment, ban users) and the ability to revert any of these actions.</p>
<p><br><b>Main Features:</b></p><ul>
<li>view post feed and comments with different listing and sort types;</li>
<li>possibility to upvote and downvote (with configurable swipe actions);</li>
<li>community and user detail (with info about moderators/moderated communities);</li>
<li>user profile with one's own posts, comments and saved items;</li>
<li>inbox with replies, mentions and direct messages;</li>
<li>global search with different result types (all, posts, comments, user, communities);</li>
<li>create and edit new posts (with optional images);</li>
<li>cross-post contents to other communities;</li>
<li>reply to post and comments (and edit replies);</li>
<li>mark posts as read (even while scrolling) and hide read contents;</li>
<li>custom appearance (color scheme, fonts, text sizes, post layout, etc.);</li>
<li>custom localization (independent of system settings);</li>
<li>block users, communities and instances (the latter requires Lemmy >= 0.19);</li>
<li>report post and comments to moderators;</li>
<li>support for multiple accounts (and multiple instances) with account-specific settings;</li>
<li>lazy scrolling (referred to as "zombie mode");</li>
<li>explore all the communities on a given instance in guest mode;</li>
<li>multi-community (community aggregation);</li>
<li>report post and comments with a reason;</li>
<li>view the moderation log;</li>
<li>community moderation tool (examine and resolve reports, ban users, feature posts, block further comments from posts, mark comments as distinguished, remove posts/comments).</li>
</ul>
```

You can copy-paste that directly, Fastlane supports HTML (and so does my repo, so does F-Droid's – and to my knowledge even PlayStore, though the former perform sanitation like stripping "dangerous things" like `<script>` while PlayStore might strip other things like `<a href` which are not used above).

As for `short_description.txt`, when removing the obvious, only the word "client" remains: the name is already in the title, it's obviously open source, etc. Maybe using this instead, as taken from the app's website:

> a mobile client for Lemmy, a federated aggregation and discussion platform

This way those who never heard about Lemmy before get a better idea (and maybe curious enough to give it a try because of that). Ok ok, yes, "mobile" is also quite obvious, but hey :wink:

Another point on Fastlane: name the `banner.png` as `featureGraphic.png` and it can be used as such. Size (1024x500) matches specs for featureGraphic, and according to my [Fastlane Cheat Sheet](https://gitlab.com/-/snippets/1895688) there is no such thing as `banner.png` (just `tvBanner.png` and that would be something different and be 1280x720).

And a last one, concerning the screenshots: they get resized here to 350px on their short size. What you have in `phoneScreenshots` are rather `sevenInchScreenshots` (or bigger). I've skipped them for now and kept the ones from Play, which are better readable then on smaller screens. You might consider moving the current ones to the "bigger place" and have screenshots from smaller screens in the default place.

I've applied the above adjustments here in my repo, so if you want to take a look:

![image](https://github.com/diegoberaldin/RaccoonForLemmy/assets/6781438/f29f2882-c51e-4e85-91ad-0eb24fa23569)
![image](https://github.com/diegoberaldin/RaccoonForLemmy/assets/6781438/7c89a063-b931-4886-86ac-12b3f727b563)

Of course the "look" will become publicly available with the next sync around 7 pm UTC :smiley:

**Comment by diegoberaldin** [comment]
Ok thank you 🙏 I'll apply the suggestions in the Gradle script for next release and update Fastlane resources as soon as possible.

**Comment by IzzySoft** [comment]
Thanks! If you let me know when the fastlane updates are in place, I'll enable them here (so they get checked and, when changed, pulled along with future releases).

**Comment by diegoberaldin** [comment]
Ok I've reviewed the descriptions and provided two new sets of updated screenshots (still a little larger than 350px in width but much smaller than the original ones).

**Comment by IzzySoft** [comment]
Thanks! Enabled it on my end. Your fulldesc is a bit shorter than mine – but it's your app, so your choice how it should be presented. Changes will be pulled with the next release then.

**Comment by diegoberaldin** [comment]
Thank you for your support and suggestions for the descriptions... being a technician and wanting to attract other developers I focused on the technologies in the early stages of the project (and forgot to change afterwards).

I've added a link to IzzyOnDroid to the README (and updated the screenshots while at it) so I think we can consider this resolved now!

Thanks again and have a good weekend! 🚀
